sonaranalyzer-dotnet/tests/SonarAnalyzer.UnitTest/Security/Ucfg/UcfgBuilder_Instructions.cs

```diff
@@ -135,31 +132,23 @@ public class Class1
         private string Property { get; set; }
         private int IntProperty { get; set; }
         private object ObjectProperty { get; set; }
-
         public string Foo(string s)
         {
             string a;
-
             Property = s;       // %0 := Namespace.Class1.Property.set [ s ]
             a = Property;       // %1 := Namespace.Class1.Property.get [  ]
                                 // a := __id [ %1 ]
-
             Property = Property;// %2 := Namespace.Class1.Property.get [  ]
                                 // %3 := Namespace.Class1.Property.set [ %2 ]
-
             Foo(Property);      // %4 := Namespace.Class1.Property.get [  ]
-                                // %5 := Namespace.Class1.Foo(string) [ %4 ]
-
+                                // %5 := Namespace.Class1.Foo(string) [ this %4 ]
             Property = Foo(Property);   // %6 := Namespace.Class1.Property.get [  ]
-                                        // %7 := Namespace.Class1.Foo(string) [ %6 ]
+                                        // %7 := Namespace.Class1.Foo(string) [ this %6 ]
                                         // %8 := Namespace.Class1.Property.set [ %7 ]
-
             ObjectProperty = s;         // %9 := Namespace.Class1.ObjectProperty.set [ s ]
-
-            ObjectProperty = 5;         // ignored
-
-            var x = IntProperty = 5;    // ignored
-
+            ObjectProperty = 5;         // %10 := Namespace.Class1.ObjectProperty.set [ const ]
+            var x = IntProperty = 5;    // %11 := Namespace.Class1.IntProperty.set [ const ]
+                                        // x := __id [ %11 ]

```

Comment:
add tests with `this.Property = s` and `other.Property = s` the correct instance should be passed

Reply:
in general all tests should include cases with `this` `base` (where needed) and `other`

Reply:
done